packages/sandcastle/src/SandcastleEditor.tsx

```diff
@@ -83,10 +83,28 @@ function SandcastleEditor({
   const {
     settings: { fontFamily, fontSize, fontLigatures },
   } = useContext(SettingsContext);
+  const documentRef = useRef(document);
   useEffect(() => {
-    internalEditorRef.current?.updateOptions({
-      fontFamily: availableFonts[fontFamily]?.cssValue ?? "Droid Sans Mono",
-    });
+    const cssName = availableFonts[fontFamily]?.cssValue ?? "Droid Sans Mono";
+    const fontFace = [...documentRef.current.fonts.values()].find(
+      (font) => font.family === cssName && font.weight === "400",
+    );
+    if (fontFace?.status !== "loaded") {
```

Comment:
For simplicity, could we instead just wait on `document.fonts.ready` and then call `monaco.editor.remeasureFonts()`?

Reply:
Per [MDN](https://developer.mozilla.org/en-US/docs/Web/API/CSS_Font_Loading_API): 

> Note that the font.load() returns a promise, so we could have handled the completion of font loading by chaining then afterwards. Using [document.fonts.ready](https://developer.mozilla.org/en-US/docs/Web/API/FontFaceSet/ready) can be better in some circumstances, as it is only called when all fonts in the document have been resolved and layout is complete.

Reply:
Ah, bummer.